modular_skyrat/modules/death_consequences_perk/death_consequences_trauma.dm

```diff
@@ -0,0 +1,524 @@
+/datum/brain_trauma/severe/death_consequences/proc/adjust_degradation(adjustment)
+	var/old_degradation = current_degradation
+	current_degradation = clamp((current_degradation + adjustment), 0, max_degradation)
+	if (current_degradation != old_degradation)
+		update_degradation_level()
+		update_effects()
+
+/datum/brain_trauma/severe/death_consequences/proc/update_degradation_level(send_reminder_if_changed = TRUE)
+	var/old_level = current_degradation_level
+	switch (current_degradation / max_degradation)
+		if (0 to 0.2)
+			current_degradation_level = DEGRADATION_LEVEL_NONE
+		if (0.2 to 0.4)
+			current_degradation_level = DEGRADATION_LEVEL_LOW
+		if (0.4 to 0.6)
+			current_degradation_level = DEGRADATION_LEVEL_MEDIUM
+		if (0.6 to 0.8)
+			current_degradation_level = DEGRADATION_LEVEL_HIGH
+		else
+			current_degradation_level = DEGRADATION_LEVEL_CRITICAL
+
+	if (send_reminder_if_changed && !final_death_delivered && (old_level != current_degradation_level))
+		send_reminder(FALSE)
+
+// EFFECTS
+
+/// Refreshes all our effects and updates their values. Kills the victim if they opted in and their degradation equals their maximum.
+/datum/brain_trauma/severe/death_consequences/proc/update_effects()
+	var/threshold_adjustment = get_crit_threshold_adjustment()
+	owner.crit_threshold = ((owner.crit_threshold - crit_threshold_currently_reduced_by) + threshold_adjustment)
+	crit_threshold_currently_reduced_by = threshold_adjustment
+
+	if (permakill_if_at_max_degradation && (current_degradation >= max_degradation))
+		and_so_your_story_ends()
+
+/// Calculates the amount that we should add to our victim's critical threshold.
+/datum/brain_trauma/severe/death_consequences/proc/get_crit_threshold_adjustment()
+	SHOULD_BE_PURE(TRUE)
+
+	var/clamped_degradation = clamp((current_degradation - crit_threshold_min_degradation), 0, crit_threshold_max_degradation)
+	var/percent_to_max = (clamped_degradation / crit_threshold_max_degradation)
+
+	var/proposed_alteration = max_crit_threshold_reduction * percent_to_max
+	var/proposed_threshold = ((owner.crit_threshold - crit_threshold_currently_reduced_by) + proposed_alteration)
+	var/overflow = max((proposed_threshold - DEATH_CONSEQUENCES_MINIMUM_VICTIM_CRIT_THRESHOLD), 0)
+	var/final_alteration = (proposed_alteration - overflow)
+
+	return final_alteration
+
+/// Ensures our victim's stamina is at or above the minimum stamina they're supposed to have.
+/datum/brain_trauma/severe/death_consequences/proc/damage_stamina(seconds_per_tick)
+	if (victim_properly_resting())
+		return
+
+	var/clamped_degradation = clamp((current_degradation - stamina_damage_minimum_degradation), 0, stamina_damage_max_degradation)
+	var/percent_to_max = min((clamped_degradation / stamina_damage_max_degradation), 1)
+	var/minimum_stamina_damage = max_stamina_damage * percent_to_max
+	// need to use a buffer because of some imprecision i noticed
```

Comment:
I noticed owner.staminaloss was a bit fucky and kept giving me stamina damage outside of the expected value, so I just expanded the range of valid stamina levels

I'll do some testing later to see if I can fix it with a round

Reply:
right, i forgot, the reason i put this in was so that your stamina damage wouldnt briefly reset due to the regen timer
the expected minimum stamina damage decreases as degradation also decreases (however, the owners stamina damgae does NOT change), meaning if i dont check for a range of values, it wont fix the regen timer issue

i believe this is necessary

Reply:
i was actually gonna do this but just    forgot